hc/api/tests/test_create_check.py

```diff
@@ -49,33 +49,95 @@ def test_it_accepts_api_key_in_header(self):
         payload = json.dumps({"name": "Foo"})
 
         ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
-
-        self.assertEqual(r['status_code'], 201)
+        response = self.post({
+            "api_key": "abc",
+            "name": "Foo"
+            })
+        self.assertEqual(response.status_code, 201)
 
     def test_it_handles_missing_request_body(self):
         ### Make the post request with a missing body and get the response
-        r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "wrong api_key")
+        response = self.post({"api_key": ""}) 
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()["error"], "wrong api_key")
 
     def test_it_handles_invalid_json(self):
-        ### Make the post request with invalid json data type
-        r = {'status_code': 400, 'error': "could not parse request body"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "could not parse request body")
+        ### Make the post
+        #  request with invalid json data type
+        response = self.client.post(self.URL,{
+            "api_key": "abc",
+            "name": "Foo"
+            })
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()["error"], "could not parse request body")
 
     def test_it_rejects_wrong_api_key(self):
-        self.post({"api_key": "wrong"},
+        responce = self.post({"api_key": "wrong"},
                   expected_error="wrong api_key")
+        self.assertEqual(responce.status_code, 400)
+        self.assertEqual(responce.json()["error"], "wrong api_key")
 
     def test_it_rejects_non_number_timeout(self):
-        self.post({"api_key": "abc", "timeout": "oops"},
+        responce = self.post({"api_key": "abc", "timeout": "oops"},
                   expected_error="timeout is not a number")
+        self.assertEqual(responce.status_code, 400)
+        self.assertEqual(responce.json()["error"], "timeout is not a number")
 
     def test_it_rejects_non_string_name(self):
-        self.post({"api_key": "abc", "name": False},
-                  expected_error="name is not a string")
+        r = self.post({
+            "api_key": "abc",
+             'name': 30,
+             "tags": "cronjob",
+             "timeout": 60000,
+             "grace": 120,
+            })
+        self.assertEqual(r.status_code, 400)
+        self.assertEqual(r.json()["error"], "name is not a string")
+
+    def test_it_can_assign_check_to_all_channels(self):
+        ### Test for the assignment of channels
```

Comment:
I like how your work has been commented out to explain bits of code. This shows adherence to PEP8 and team conventions. However, to make your code neat you could remove the extra `#` since you only need one for a comment.